controllers/openstackdataplanerole_controller.go

```diff
@@ -68,9 +78,190 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
+		}
+	}
+
+	_, err = r.GenerateInventory(ctx, instance, nodes.Items)
+	if err != nil {
+		util.LogErrorForObject(helper, err, fmt.Sprintf("Unable to generate inventory for %s", instance.Name), instance)
+		return ctrl.Result{}, err
+	}
+
 	return ctrl.Result{}, nil
 }
 
+// GenerateInventory yields a parsed Inventory
+func (r *OpenStackDataPlaneRoleReconciler) GenerateInventory(ctx context.Context, instance *dataplanev1beta1.OpenStackDataPlaneRole, nodes []dataplanev1beta1.OpenStackDataPlaneNode) (string, error) {
+	var (
+		err      error
+		hostName string
+	)
+
+	inventory := ansible.MakeInventory()
+	overcloud := inventory.AddGroup("allovercloud").AddChild(ansible.MakeGroup("overcloud"))
+
+	if instance.Spec.NodeTemplate.AnsibleUser != "" {
+		overcloud.Vars["ansible_user"] = instance.Spec.NodeTemplate.AnsibleUser
+	}
+	if instance.Spec.NodeTemplate.AnsiblePort > 0 {
+		overcloud.Vars["ansible_port"] = instance.Spec.NodeTemplate.AnsiblePort
+	}
+	if instance.Spec.NodeTemplate.Managed {
+		overcloud.Vars["managed"] = instance.Spec.NodeTemplate.Managed
+	}
+	if instance.Spec.NodeTemplate.ManagementNetwork != "" {
+		overcloud.Vars["management_network"] = instance.Spec.NodeTemplate.ManagementNetwork
+	}
+	if instance.Spec.NodeTemplate.NetworkConfig.Template != "" {
+		overcloud.Vars["network_config"] = fmt.Sprintf("{template: %s}", instance.Spec.NodeTemplate.NetworkConfig.Template)
+	}
+	if len(instance.Spec.NodeTemplate.Networks) > 0 {
+		var network string // the resulting string containing each network
+		for _, netMap := range instance.Spec.NodeTemplate.Networks {
+			if netMap.FixedIP != "" && netMap.Network == "" {
+				network += fmt.Sprintf("{%s: %s},", "fixedIP", netMap.FixedIP)
+			}
+			if netMap.FixedIP != "" && netMap.Network != "" {
+				network += fmt.Sprintf("{%s: %s, %s: %s},",
+					"fixedIP", netMap.FixedIP, "network", netMap.Network)
+			}
+		}
+		overcloud.Vars["networks"] = fmt.Sprintf("[%s]", strings.TrimSuffix(network, ","))
+	}
+
+	ansibleVarsData := make(map[string]interface{})
+	err = yaml.Unmarshal([]byte(instance.Spec.NodeTemplate.AnsibleVars), ansibleVarsData)
+	if err != nil {
+		return "", err
+	}
+	for key, value := range ansibleVarsData {
+		overcloud.Vars[key] = value
+	}
+
+	compute := overcloud.AddChild(ansible.MakeGroup("Compute"))
+	for _, node := range nodes {
+		host := compute.AddHost(node.Name)
+		if node.Spec.Node.AnsibleUser != "" {
+			host.Vars["ansible_user"] = node.Spec.Node.AnsibleUser
+		}
+		if node.Spec.Node.AnsiblePort > 0 {
+			host.Vars["ansible_port"] = node.Spec.Node.AnsiblePort
+		}
+		if node.Spec.Node.Managed {
+			host.Vars["managed"] = node.Spec.Node.Managed
+		}
+		if node.Spec.Node.ManagementNetwork != "" {
+			host.Vars["management_network"] = node.Spec.Node.ManagementNetwork
+		}
+		if node.Spec.Node.NetworkConfig.Template != "" {
+			host.Vars["network_config"] = fmt.Sprintf("{template: %s}", node.Spec.Node.NetworkConfig.Template)
+		}
+		if len(node.Spec.Node.Networks) > 0 {
```

Comment:
Since lines 160-170 and 200-210 are basically the same, do you want to define a separate function and call it instead?

Reply:
yes, I'm planning to do it